File: src/oracle/OracleWeight.py
from dataclasses import dataclass


@dataclass
class OracleWeight:
    coverage: float = 1.0
    diversity: float = 1.0
    specificity: float = 1.0
    interestingness: float = 1.0
    conciseness: float = 1.0

    def to_dict(self) -> dict[str, float]:
        return {
            "coverage": self.coverage,
            "diversity": self.diversity,
            "specificity": self.specificity,
            "interestingness": self.interestingness,
            "conciseness": self.conciseness,
        }

File: client/src/controller/oracleWeight.ts
import { signal } from '@preact/signals-react';
import { OracleWeight } from '../types/OracleWeight';

const weightSignal = signal<OracleWeight>({
    specificity: 1,
    interestingness: 1,
    coverage: 1,
    diversity: 1,
    conciseness: 1,
});



export { weightSignal };
